app/src/main/java/de/westnordost/streetcomplete/osm/street_parking/StreetParking.kt

```diff
@@ -81,19 +81,32 @@ private val ParkingPosition.estimatedWidthOnRoadFactor: Float get() = when (this
 fun LeftAndRightStreetParking.applyTo(tags: Tags) {
     val currentParking = createStreetParkingSides(tags)
 
-    // was set before and changed: may be incorrect now - remove subtags!
-    if (currentParking?.left != null && currentParking.left != left ||
-        currentParking?.right != null && currentParking.right != right) {
-        /* This includes removing any parking:condition:*, which is a bit peculiar because most
-         * values are not even set in this function. But on the other hand, when the physical layout
-         * of the parking changes (=redesign of the street layout and furniture), the condition may
-         * very well change too, so better delete it to be on the safe side. (It is better to have
-         * no data than to have wrong data.) */
-        val parkingLaneSubtagging = Regex("^parking:(lane|condition):.*")
-        for (key in tags.keys) {
-            if (key.matches(parkingLaneSubtagging)) {
-                tags.remove(key)
+    // parking:condition:<left/right/both>
+    val conditionRight = right?.toOsmConditionValue()
+    val conditionLeft = left?.toOsmConditionValue()
+
+    /* Clean up previous parking tags when the associated side is changed. This can include removing
+     * parking:condition:*, which is a bit peculiar because most values are not even set in this
+     * function. But on the other hand, when the physical layout of the parking changes (=redesign
+     * of the street layout and furniture), the condition may very well change too, so better delete
+     * it to be on the safe side. (It is better to have no data than to have wrong data.) */
+    val keysToRemove = buildList {
```

Comment:
In general I like to avoid such Kotlin constructs that implicitly change what `this` is because it adds a layer of confusion and a source of error (you need the IDE to help you display what `this` is or know the documentation by heart)